models/solution/route.rb

```diff
@@ -77,6 +115,19 @@ def shift_route_times(shift_amount, shift_start_index = 0)
         }
         self.info.end_time += shift_amount if self.info.end_time
       end
+
+      def shift_route_distance(shift_amount, shift_start_index = 0)
+        return if shift_amount == 0
+
+        raise 'Cannot shift the route, there are not enough stops' if shift_start_index > self.stops.size
+
+        self.info.total_distance += shift_amount
+        self.stops.each.with_index{ |stop, index|
+          next if index <= shift_start_index
+
+          stop.info.current_distance += shift_amount
```

Comment:
A check against too much shift?
```suggestion
          next if index <= shift_start_index

          raise 'Cannot shift the route, shift_amount is too small -- it would lead to negative current_distance` if shift_amount < -stop.info.current_distance

          stop.info.current_distance += shift_amount
```